kubetest/kops.go

```diff
@@ -487,3 +498,47 @@ func (k *kops) runKopsDump() (*kopsDump, error) {
 
 	return dump, nil
 }
+
+// kops deployer implements publisher
+var _ publisher = &kops{}
+
+// Publish will publish a success file, it is called if the tests were successful
+func (k kops) Publish() error {
+	if k.kopsPublishVersion == "" {
+		return nil
+	}
+
+	if k.kopsVersion == "" {
+		return errors.New("kops-version not set; cannot publish")
+	}
+
+	return xmlWrap("Publish kops version", func() error {
+		log.Printf("Set %s version to %s", k.kopsPublishVersion, k.kopsVersion)
+		return gcsWrite(k.kopsVersion, []byte(k.kopsPublishVersion))
+	})
+}
+
+// gcsWrite uploads contents to the dest location in GCS.
+// It currently shells out to gsutil, but this could change in future.
+func gcsWrite(dest string, contents []byte) error {
```

Comment:
Seems like a dup of https://github.com/kubernetes/test-infra/blob/master/kubetest/main.go#L1011? Can we colapse these perhaps?

Reply:
It's not a total dup, in that contents already exists on disk in main.go's publish function.  But I can do that (I'll move this func to utils) - for kubernetes publish it would mean that we read the version file, write it out to a new temp file, and then `gsutil cp` that.

But on the plus side if we have gcsWrite call the GCS API directly, the temp file (and most of the complexity goes away).

Reply:
Various PRs in flight (like Steve's) vendor gcs which will make this possible (woo!)